plans/phase-1/final-cr-definition.md

```diff
@@ -0,0 +1,922 @@
+# This 'Cell' CR is created from an item in the `spec.cells.templates` list
+# in the parent `MultigresCluster`.
+#
+# This specific example is for 'us-east-1a', which had no 'topoServer'
+# block, so it defaults to using the 'global' topoServer.
+#
+apiVersion: multigres.com/v1alpha1
+kind: Cell
+metadata:
+  name: "example-multigres-cluster-us-east-1a"
+  namespace: example
+  labels:
+    multigres.com/cluster: "example-multigres-cluster"
+    multigres.com/cell: "us-east-1a"
+  ownerReferences:
+  # The Cell CR is owned by the MultigresCluster
+  - apiVersion: multigres.com/v1alpha1
+    kind: MultigresCluster
+    name: "example-multigres-cluster"
+    uid: "a1b2c3d4-1234-5678-90ab-f0e1d2c3b4a5"
+    controller: true
+spec:
+  # The logical name of the cell, copied from the template.
+  name: "us-east-1a"
+
+  # The parent MultigresCluster passes down the relevant
+  # images for this controller to use.
+  images:
+    multigateway: "multigres/multigres:latest"
+    multiorch: "multigres/multigres:latest"
+
+  # The multigateway spec is copied from multigrescluster
+  multigateway:
+    replicas: 2
+    resources:
+      requests:
+        cpu: "500m"
+        memory: "512Mi"
+      limits:
+        cpu: "2"
+        memory: "1Gi"
+
+ # The multiorch spec is copied from multigrescluster.
+  multiorch:
+    replicas: 1
+    resources:
+      requests:
+        cpu: "100m"
+        memory: "128Mi"
+       limits:
+        cpu: "200m"
+        memory: "256Mi"
+  
+  # A reference to the GLOBAL TopoServer.
+  # This is always populated by the parent controller.
+  globalTopoServer:
+    rootPath: "/multigres/global"
+    clientServiceName: "example-multigres-cluster-global-client"
+
+  # ALTERNATIVE CONFIG: (Using the Global TopoServer)
+  #
+  # Because the 'us-east-1a' cell in the parent CR had no 'topoServer'
+  # block, the MultigresCluster controller sets this to empty.
+  # topoServer: {}
+
+  # ALTERNATIVE CONFIG: Inline Definition (External)
+  # If this were 'us-east-1a', the MultigresCluster controller would
+  # have copied the 'external' block directly, like this:
+  #
+  # topoServer:
+  #   external:
+  #     address: "etcd-us-east-1a.my-domain.com:2379"
+  #     rootPath: "/multigres/us-east-1a"
+
+  # ALTERNATIVE CONFIG: (Managed Local) ---
+  # If this were 'us-east-1b', the MultigresCluster controller would
+  # have copied the 'managedSpec' block directly, like this.
+  # The Cell controller would then be responsible for creating
+  # a NEW 'TopoServer' CR from this spec, with this Cell
+  # as its owner.
+  #
+  # topoServer:
+  #   managedSpec:
+  #     rootPath: "/multigres/us-east-1b"
+  #     image: "quay.io/coreos/etcd:v3.5.17"
+  #     replicas: 1
+  #     dataVolumeClaimTemplate:
+  #       accessModes: ["ReadWriteOnce"]
+  #       resources:
+  #         requests:
+  #           storage: "5Gi"
+
+  # List of all cells in the cluster for discovery.
+  allCells:
+  - "us-east-1a"
+  - "us-east-1b"
+  - "us-east-1c"
```

Comment:
This cell information will be propagated to many cells -- this is something I'd like to question the Multigres team later whether this is something we need to carry over in the CRs.